### DISCUSSION TITLE
Opening "nested" apps

### DISCUSSION THREAD

**Discussion by gregkos** [rxhanson/Charmstone-Community#40] [Q&A] [answered]
Let me preface this by saying that I'm fairly confident "nested" is not the correct term, but I do not know how to phrase this in a way that fits the title. Ignoring that, here's a more thorough explanation. I will be talking about a specific set of apps here, but I would assume this is not a unique setup.

I am working as a developer and a good chunk of my development work is being done through various JetBrains IDEs. For a long time, JetBrains has been offering the Toolbox, an easy way to manage the installations of different IDEs on your system. In the past year or so, they have made a few changes in the way the standalone apps are installed, and they appear as a collection "inside" the Toolbox instead of separate apps inside `/Applications`.

As such, Charmstone appears to be having trouble finding those apps to select as a switch target. At the same time, other "launchers", such as Alfred, have no issue understanding and retrieving the correct `.app` files using fuzzy searching. I have very limited experience developing for Mac OS so I might be comparing apples to oranges, but I guess there is *some* way to teach Charmstone where to look. Also, while the JetBrains suite is the most prominent example I can think of, I doubt they're the only ones bundling their apps in this way, so support for that might benefit other groups of users too.

If there's anything I can share that would help investigating this, I'd be happy to. If there's an obvious solution to this I have completely missed, I apologize in advance and I'd be delighted to know what I should do to have them show up.

Thank you!

**Comment by rxhanson** [comment, marked as answer]
If I understand correctly, you can do one of two things:
1. Open Finder and drag and drop the app you want into the Charmstone's preferences window in the place that you want.
2. Select Open File or Folder in Charmstone's preferences window when selecting a space you want to add and browse to the App that you want. (This will point to the location, rather than the identifier of the application).

Let me know if that's not what you were looking for.

**Reply by gregkos** [reply]
I tried solution #1 and it indeed worked!

However, while trying it, I realized the place those `.app` files are is `~/Applications` (as opposed to `/Applications`) and indeed there are other apps in there as well. Perhaps a way to include that folder (which seems fairly standard) would be a nice addition to the app?

**Reply by rxhanson** [reply]
Thanks for the followup. It actually is supposed to search that directory, but there must be a bug. I'll add it to my backlog.

**Reply by gregkos** [reply]
I'm on Sonoma, if that matters. Perhaps a permissions issue with the new OS.
